File: resiliences-service/Payloads/CreateTargetSavingsRequest.cs
using System.ComponentModel.DataAnnotations;

namespace resiliences_service.Payloads
{
    public class CreateTargetSavingsRequest
    {
        [Required]
        public long UserId { get; set; }

        [Required]
        public long WalletId { get; set; }

        [Required]
        [StringLength(10)]
        public string CurrencyCode { get; set; } = "NGN";

        [Required]
        [StringLength(100, MinimumLength = 1)]
        public string GoalName { get; set; } = default!;

        [StringLength(500)]
        public string? Description { get; set; }

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Target amount must be greater than 0")]
        public decimal TargetAmount { get; set; }

        public DateTime? TargetDate { get; set; }

        [StringLength(10)]
        public string? GoalIcon { get; set; }
    }
}

File: domain/src/main/java/com/epay/domain/investment/input/CreateTargetSavingsRequest.java
package com.epay.domain.investment.input;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class CreateTargetSavingsRequest {

    @NotNull
    private Long userId;

    @NotNull
    private Long walletId;

    @NotNull
    @Size(max = 10)
    private String currencyCode = "NGN";

    @NotBlank
    @Size(max = 100)
    private String goalName;

    @Size(max = 500)
    private String description;

    @NotNull
    @DecimalMin(value = "0.01", message = "Target amount must be greater than 0")
    private BigDecimal targetAmount;

    private LocalDateTime targetDate;

    @Size(max = 10)
    private String goalIcon;
}
